### DISCUSSION TITLE
autoUnsubcribe operator - which makes auto unscribe from an observable automatic

### DISCUSSION THREAD

**Discussion by hansschenker** [ReactiveX/rxjs#7486] [Ideas / Feature request]
import { autoUnsubscribe } from './autoUnsubscribe';

describe('autoUnsubscribe operator', () => {
  it('should unsubscribe from source when all subscribers unsubscribe', () => {
    const source$ = of('a', 'b', 'c');
    const autoUnsubscribed$ = source$.pipe(autoUnsubscribe());

    const sub1 = autoUnsubscribed$.subscribe();
    const sub2 = autoUnsubscribed$.subscribe();

    sub1.unsubscribe();
    sub2.unsubscribe();

    expect(sub1.closed).toBe(true);
    expect(sub2.closed).toBe(true);
    // Additional assertions for source subscription can be added
  });
});

**Comment by AronHoxha** [comment]
I believe you can achieve this by using `share`/`shareReplay`  with refcount option

**Comment by hansschenker** [comment]
yes you can but you have to know it and it needs additional steps, this
operator would help not to forget to unsubscribe
